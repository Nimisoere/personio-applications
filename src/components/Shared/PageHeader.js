import React from "react";
import { Container } from "reactstrap";
import { Seo } from "../Shared";
import PropTypes from "prop-types";

import { seoObject } from "../../_constants";
import "./PageHeader.scss";

export const PageHeader = ({ pageId, pageTitle, isBase }) => (
  <div className="page-header">
    <Container>
      <Seo
        title={
          seoObject[pageId]
            ? seoObject[pageId].title || seoObject.base.title
            : seoObject.base.title
        }
        description={
          seoObject[pageId]
            ? seoObject[pageId].description || seoObject.base.description
            : seoObject.base.description
        }
        base={isBase}
      />
      <h4>{pageTitle}</h4>
    </Container>
  </div>
);

PageHeader.propTypes = {
  pageId: PropTypes.string,
  pageTitle: PropTypes.string,
  isBase: PropTypes.bool
};
