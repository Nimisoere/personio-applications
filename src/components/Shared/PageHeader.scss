.page-header {
  padding: 2rem 0;
  background: #d5f0f7;
  h4 {
    font-size: 1.1rem;
    font-weight: bold;
  }
}
