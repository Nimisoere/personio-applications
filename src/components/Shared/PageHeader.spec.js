import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { PageHeader } from "./PageHeader";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    pageId: "test",
    pageTitle: "test",
    isBase: false
  };
  const enzymeWrapper = shallow(<PageHeader {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("components", () => {
  describe("Page Header", () => {
    it("should render", () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
