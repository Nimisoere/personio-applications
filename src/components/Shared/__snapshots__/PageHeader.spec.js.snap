// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Page Header should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader
    isBase={false}
    pageId="test"
    pageTitle="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Seo
          base={false}
          description="Personio Applications"
          title="Personio Applications"
        />,
        <h4>
          test
        </h4>,
      ],
      "className": "page-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "base": false,
          "description": "Personio Applications",
          "title": "Personio Applications",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Seo
            base={false}
            description="Personio Applications"
            title="Personio Applications"
          />,
          <h4>
            test
          </h4>,
        ],
        "className": "page-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "base": false,
            "description": "Personio Applications",
            "title": "Personio Applications",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
